server/routes/index.ts

```diff
@@ -74,6 +78,66 @@ router.get('/health', async (req, res) => {
   res.send('OK').end()
 })
 
+const substackDomainCache = {
+  // uncomment the below for test purpose
+  'localhost:3100': 'polymorpher.substack.com'
+}
+
+const client = buildClient()
+
+router.get('/substack/api/v1/archive',
+  limiter(),
+  async (req, res) => {
+    const host = req.get('host')
+
+    if (!host) {
+      return
+    }
+
+    let substackDomain = substackDomainCache[host]
+
+    if (!substackDomain) {
+      const [subdomain, sld] = host?.split('.')
+
+      substackDomain = await client.getLandingPage(sld, subdomain)
+
+      if (!substackDomain.endsWith('.substack.com')) {
+        throw new Error('Not substack page')
+      }
+
+      substackDomainCache[host] = substackDomain
+    }
+
+    const { data } = await axiosBase.get(`https://${substackDomain}/api/v1/archive`, { params: req.query })
+
+    res.status(200).send(data)
+  }
+)
+
+const axiosBase = axios.create({ timeout: 15000 })
+
+router.get('/substack',
+  limiter(),
+  async (req, res) => {
+    try {
```

Comment:
this should restrict resources under the substack site set for landingPage under SLD/subdomain, otherwise it becomes a general proxy

Reply:
Done